File: java/gradle/src/gatling/java/io/gatling/grpc/demo/Configuration.java
package io.gatling.grpc.demo;

import io.gatling.javaapi.grpc.GrpcProtocolBuilder;

import static io.gatling.javaapi.grpc.GrpcDsl.grpc;

public class Configuration {

    private Configuration() {
        // Do nothing.
    }

    public static GrpcProtocolBuilder baseGrpcProtocol(String host, int port) {
        return grpc.forAddress(host, port).useInsecureTrustManager();
    }

    public static GrpcProtocolBuilder baseGrpcProtocolWithMutualAuth(String host, int port) {
        return grpc.forAddress(host, port)
            .channelCredentials("#{channelCredentials}")
            .overrideAuthority("gatling-grpc-demo-test-server");
    }
}

File: kotlin/gradle/src/gatling/kotlin/io/gatling/grpc/demo/Configuration.kt
package io.gatling.grpc.demo

import io.gatling.javaapi.grpc.GrpcDsl
import io.gatling.javaapi.grpc.GrpcProtocolBuilder

object Configuration {
  @JvmStatic
  fun baseGrpcProtocol(host: String, port: Int): GrpcProtocolBuilder {
    return GrpcDsl.grpc.forAddress(host, port).useInsecureTrustManager()
  }
}
